File: src/com/mjie/other/Test1.java
package com.mjie.other;

import java.util.Arrays;
import java.util.List;

/**
 * @author panmingjie
 * @date 2018/12/22 14:42
 */
public class Test1 {
    public static void main(String[] args) {
        List<String> filter1 = Test1Kt.myFilter(Arrays.asList("hello", "world"));
        List<Integer> filter2 = Test1Kt.myFilter2(Arrays.asList(10, 20));
        System.out.println(filter1);
        System.out.println(filter2);
    }
}

File: src/com/mjie/other/Test1.kt
package com.mjie.other

fun List<String>.myFilter(): List<String> {
    return listOf("hello", "world")
}

@JvmName("myFilter2")
fun List<Int>.myFilter(): List<Int> {
    return listOf(1, 2)
}

fun main(args: Array<String>) {
    var str = listOf<String>("hello")
    val myFilter = str.myFilter()
    println(myFilter)
}